src/openforms/submissions/tests/test_models.py

```diff
@@ -617,3 +620,20 @@ def test_clear_execution_state_without_execution_state(self):
         self.assertFalse(hasattr(submission, "_execution_state"))
 
         submission.clear_execution_state()
+
+
+@temp_private_root()
+class PDFSubmissionReportTests(HypothesisTestCase):
+    @given(
+        st.text(
+            min_size=1,
+            alphabet=st.characters(blacklist_characters="\x00", codec="utf-8"),
+        )
+    )
+    # see ticket #3470
+    @settings(deadline=500)
+    def test_names_donot_break_pdf_generation(self, form_name):
+        report = SubmissionReportFactory.create(submission__form__name=form_name)
+        report.generate_submission_report_pdf()
+
+        self.assertTrue(report.content)
```

Comment:
might even make it more obvious that the file exists on disk:

```py
self.assertTrue(report.content.storage.exists(report.content.name))
```
`report.content` (if you stringify it) only returns the path where it should be, but that doesn't guarantee yet that it exists on disk.